src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -142,6 +142,30 @@ internal class AnalyzerManager
         }
 
         /// <summary>
+        /// Returns true if analyzer registered a compilation start action during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/>
+        /// which registered a compilation end action and at least one other analyzer action, that the end action depends upon.
+        /// </summary>
+        public async Task<bool> GetAnalyzerHasDependentCompilationEndAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
AnalyzerManager can never end up as public type as AnalyzerActions go along with it. What worries me more about your approach is having 2 separate code pieces for doing the same work. We already found 2 subtle issues around crash due to TaskCancelledException and DeclarePublicAPIAnalyzer perf issue in this piece of code. Having this computation at 2 places to have a cleaner API for an internal type, doesn't appeal to me. I can add a comment saying this API must go away once we moved to a shared driver or any of AnalyzerDriver related APIs go public?